### PR TITLE
Handle intentionally empty grpc BestOptions

### PR DESCRIPTION
**Pull request by rrangith**
#### What type of PR is this?

/kind bug

## Problem
When trying to filter out all options from the gRPC expander side, I noticed that the gRPC expander client on cluster autoscaler would return all the expansion options, rather than none of them.

This is a problem because in some cases I want to filter out all options on the gRPC expander server side, however the cluster-autoscaler client is unfiltering the options when the server returns nothing. 

I would expect that if the gRPC expander server returns no best options, CA should fail its scaleup since it has no options available.

## Current Behaviour

In the case of nil best options returned from the gRPC expander, the gRPC client returns all original options https://github.com/kubernetes/autoscaler/blob/721ea01f7f284484fa491ae89e08bded434bcd46/cluster-autoscaler/expander/grpcplugin/grpc_client.go#L95-L98

## Proposed Change

If the grpc expander server returns nil, the client should respect that and also return nil. This is similar to what other expanders do https://github.com/kubernetes/autoscaler/blob/721ea01f7f284484fa491ae89e08bded434bcd46/cluster-autoscaler/expander/waste/waste.go#L67-L69

This change does break things for users who rely on a nil response from the grpc expander server, however this behaviour is very unexpected and I believe it should change to be more predictable

If users want the original behaviour where the client ends up returning all original options, then on the gRPC expander server side they can return an error. Then expander will apply no filtering and return the original expansionOptions.

When the scaleup orchestrator gets an empty best option, it will fail the scaleup as expected https://github.com/kubernetes/autoscaler/blob/721ea01f7f284484fa491ae89e08bded434bcd46/cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go#L178-L185

## Testing

I deployed cluster-autoscaler with this change on my cluster and confirmed it does not scaleup when no options are returned, as expected.

#### Does this PR introduce a user-facing change?

```release-note
If the gRPC expander server returns nil for its best options, the gRPC expander client will return nil
```

### CODE REVIEW

File: cluster-autoscaler/expander/grpcplugin/grpc_client_test.go [hunk 1]
```diff
@@ -263,13 +286,15 @@ func TestBestOptionsErrors(t *testing.T) {
...
 	for _, tc := range testCases {
 		grpcNodeInfoMap := populateNodeInfoForGRPC(tc.nodeInfo)
-		mockClient.EXPECT().BestOptions(
-			gomock.Any(), gomock.Eq(
-				&protos.BestOptionsRequest{
-					Options: []*protos.Option{&grpcEoT2Micro, &grpcEoT2Large, &grpcEoT3Large, &grpcEoM44XLarge},
-					NodeMap: grpcNodeInfoMap,
-				})).Return(&tc.mockResponse, tc.errResponse)
-		resp := g.BestOptions(options, tc.nodeInfo)
+		if tc.client.grpcClient != nil {
```
Comment (MaciekPytel) [inline cluster-autoscaler/expander/grpcplugin/grpc_client_test.go:289]: Why is this needed? It seems like if this happen, the test case haven't been properly initialized, in which case it should fail and not pass silently

Comment (rrangith) [reply cluster-autoscaler/expander/grpcplugin/grpc_client_test.go:289]: @MaciekPytel  it was for this test case https://github.com/kubernetes/autoscaler/blob/29de5009060c12ef95d70eaacfdc6bc83c151524/cluster-autoscaler/expander/grpcplugin/grpc_client_test.go#L259-L263, prior to my change, the test was not executing the correct code path since `tc.client` was never being used, it was always `g` being used as the client https://github.com/kubernetes/autoscaler/pull/7351/files#diff-5392da3174a0a9693626803662fd3c6871211cfe739ec3b7d6a5efff38e1cae9L272
